src/protocols/secure_channel/CASESession.h

```diff
@@ -237,16 +237,84 @@ class DLL_EXPORT CASESession : public Messaging::UnsolicitedMessageHandler,
         size_t encrypted2Length = 0;
         const ReliableMessageProtocolConfig * responderMrpConfig;
     };
+    struct ParsedSigma2
+    {
+        ByteSpan responderRandom;
+        uint16_t responderSessionId;
+        ByteSpan responderEphPubKey;
+        Platform::ScopedMemoryBufferWithSize<uint8_t> msgR2Encrypted;
+        bool responderSessionParamStructPresent = false;
+        SessionParameters responderSessionParams;
+    };
+
+    struct ParsedSigma2TBEData
+    {
+        ByteSpan responderNOC;
+        ByteSpan responderICAC;
+        Crypto::P256ECDSASignature tbsData2Signature;
+        ByteSpan resumptionId;
```

Comment:
For all the bytespans, the backing of the bytespan and lifetime must be indicated.